### DISCUSSION TITLE
Zen Cart 2.0.x

### DISCUSSION THREAD

**Discussion by scottcwilson** [zencart/zencart#5892] [General]
This thread is intended to cover all things 2.0.0.

A roadmap thread is on the forum at 

https://www.zen-cart.com/showthread.php?229610-The-next-release-Zen-Cart-2-0-0

This post will go into more detail to provide developer-level information. 

## Version number change

Zen Cart is moving to semantic versioning, which is we called this release 2.0.0.  [Read more about how to interpret Zen Cart version numbers](https://docs.zen-cart.com/user/about_us/versioning/).

The version number change is not intended to convey a significant re-architecting or addition of large number of features; it was done to get us aligned with version numbering standards. 

## PHP Version

Zen Cart 2.0.0 requires a minimum PHP version of 8.0.  See 

https://docs.zen-cart.com/user/first_steps/server_requirements/#php-version

## Language File Changes

Zen Cart 2.0.0 is built on top of Zen Cart 1.5.8, so it uses the newer style language files.  If you are upgrading from a release prior to 1.5.8, see 

https://docs.zen-cart.com/release/whatsnew_2.0.0#arrayBasedLanguageFiles

for guidance on updating your language files.

## Documentation 

Documentation on Zen Cart 2.0.0 is now on documentation site.

https://docs.zen-cart.com/user/upgrading/release_specific_upgrade_considerations/#zen-cart-200

The What's New file entries can be seen here: 

https://docs.zen-cart.com/release/whatsnew_2.0.0.html

**Comment by drbyte** [comment]


**Reply by drbyte** [reply]
UPDATE: 
Now that 2.0.0 is released, and we're using semantic versioning conventions:

- `master` continues to be the branch where all new features should be PR'd to (and will become 2.1 when it comes time to tag that)
- `2.0` is the branch where all bugfixes for 2.0 should be PR'd to. (And we will re-merge `2.0` over to `master` after merging the PR into 2.0, so that `master` remains consistent)

When 2.1 gets released, then:
- `2.1` is the branch where all bugfixes for 2.1 should be PR'd to.
- `master` will be for future such as 2.2
- `2.0` branch will still be used for bugfixes specific to 2.0.x

etc.

**Comment by drbyte** [comment]
UPDATE: 
Now that 2.0.0 is released, and we're using semantic versioning conventions:

- `master` continues to be the branch where all new features should be PR'd to (and will become 2.1 when it comes time to tag that)
- `2.0` is the branch where all bugfixes for 2.0 should be PR'd to. (And we will re-merge `2.0` over to `master` after merging the PR into 2.0, so that `master` remains consistent). 
- (so, developers wanting to submit a bugfix to 2.0, should check out the 2.0 branch in order to PR against it)

When 2.1 gets released, then:
- `2.1` is the branch where all bugfixes for 2.1 should be PR'd to.
- `master` will be for future such as 2.2
- `2.0` branch will still be used for bugfixes specific to 2.0.x

etc.

**Reply by scottcwilson** [reply]
Developers: You will have to create the 2.0 branch on your fork.  In the Github UI, from the code tab, click the branch dropdown, then View All Branches.   Click the "New branch" button on the right, and select Source "zencart/zencart" and branch "2.0", and name your new branch "2.0", then click the "Create new branch" button.

**Reply by scottcwilson** [reply]
Our intention is to merge bug fixes only into 2.0 and new feature content (and non-fix updates) into 2.1. so that 2.0.x updates are easier and smaller for store owners.

**Comment by scottcwilson** [comment]
Bug Fixes and Improvements in Coming 2.x.y Releases will be posted here: 

https://docs.zen-cart.com/release/work_in_progress

until they migrate into a whatsnew_*.html page closer to the release.

**Comment by scottcwilson** [comment]
Release 2.0.1 is coming soon!
